source/filters/localizeStringDates/localizeStringDates.filter.ts

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+import * as angular from 'angular';
+import * as moment from 'moment';
+
+import { services } from 'typescript-angular-utilities';
+import __date = services.date;
+import __object = services.object;
+import __timezones = services.timezone;
+
+
+export var moduleName: string = 'rl21.ui.filters.localizeStringDates';
+export var filterName: string = 'rlLocalizeStringDates';
+
+export interface ILocalizeStringDates {
+	(input?: string): string;
+}
+
+/**
+ *Filters a string and find all, datetimes that match the format M/D/YYYY H:M:S AM|PM that is used to store dates in message log body text.
+* Assumes they are UTC  and parses them, converts to local browser time replaces them in the string.
+ */
+export function localizeStringDates(): ILocalizeStringDates {
+	'use strict';
+	return (input?: string): string => {
+		if (input == null) {
+			return '';
+		}
+
+		//if no timezone is specified then use moment to get the browsers time zone.
+		const timezone: string = __timezones.timezoneService.currentTimezone.momentName;
+		const regex: RegExp = /([1-9]|1[0-2])\/([1-9]|1\d|2\d|3[01])\/(19|20)\d{2} ([1-9]|1[0-9])\:([0-9])([0-9])(?:\:([0-9])([0-9]))? (AM|PM)(?: UTC)?/g;
```

Comment:
The main advantage would just be splitting up the different segments of the expression into named variables explaining their purpose, such as 'years', 'months', 'days', 'hours', 'minutes', etc, so someone can see at a glance what's going on. I basically look at that expression and then quickly look away, hoping that it does what it's supposed to. But I suppose being able to copy and paste it into regex101.com is an advantage.